custom/conf/app.example.ini

```diff
@@ -2478,3 +2478,18 @@ ROUTER = console
 ;PROXY_URL =
 ;; Comma separated list of host names requiring proxy. Glob patterns (*) are accepted; use ** to match all hosts.
 ;PROXY_HOSTS =
+
+; [bots]
+;; Enable/Disable bots capabilities
+;ENABLED = false
+;DEFAULT_BOTS_URL = https://gitea.com
```

Comment:
What is the use of a default value for `DEFAULT_BOTS_URL`?
Shouldn't it be unset to force the user to point to its own bots?

Or do you intend to provide some bots for free on gitea.com?

Reply:
Yes, currently most of them are mirrored from GitHub and will be changed when necessary. All of them are licensed on MIT. To make it usable out of box, the default value is set. Users could copy/mirror these actions in any git service he can access and change the value to that.

Reply:
I think there are some misunderstandings, "provide some bots for free" sounds like providing some CICD runners for free, but that's not true yet.

`DEFAULT_BOTS_URL` indicates where to get [actions used in steps](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions), and the default value `https://gitea.com/` means `https://gitea.com/actions`, instead of `https://github.com/actions`.

 According to [the vote result](https://github.com/go-gitea/gitea/issues/13539#issuecomment-1331930126), it'll be rename to `DEFAULT_ACTIONS_URL`, I think it might help to understand it better.

Reply:
By the way, the current design is that runners will connect to Gitea instance initiatively, so the Gitea instance doesn't need to know the address of runners.

Reply:
Thanks for the reply, I understand a bit better. I still have a couple of questions

Is this default URL really needed, or wouldn't it be better to force the users to specify the full URL like in go packages ?
If you prefer to offer a "github like" experience, then why not use github.com as default?

Reply:
> Imho all this configuration should really go into agent/bot/runner side not in gitea

But when you want to change these properties, you have to change them from runner one by one.

Reply:
> > We will support a full URL uses like uses: selfhost-gitea.com/actions/checkout@v3
> 
> ~Now I understand that using actions/runner as an agent is not feasible~(even this can be implemented with an unmodified actions/runner). I made an [agent](https://github.com/christopherhx/github-act-runner) based on nektos/act, which uses the actions/runner protocol instead of creating a protocol like done with gitea/act_runner.
> 
> > DEFAULT_BOTS_URL indicates where to get [actions used in steps](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions), and the default value https://gitea.com/
> 
> This explains why I couldn't access actions from my own gitea instance. I would expect the default is the current gitea instance, if the actions isn't mirrored look in gitea.com, then github.com ( Preferable a configurable list of instances to look for actions).

If the default value is his own gitea instance, he have to mirror some actions before he can test something. I think we should let new users could use it easily and then when he becomes a familiar user of Gitea he can decide to mirror something and change it.

Reply:
> > We will support a full URL uses like uses: selfhost-gitea.com/actions/checkout@v3
> 
> ~Now I understand that using actions/runner as an agent is not feasible~(even this can be implemented with an unmodified actions/runner). I made an [agent](https://github.com/christopherhx/github-act-runner) based on nektos/act, which uses the actions/runner protocol instead of creating a protocol like done with gitea/act_runner.
> 
> > DEFAULT_BOTS_URL indicates where to get [actions used in steps](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions), and the default value https://gitea.com/
> 
> This explains why I couldn't access actions from my own gitea instance. I would expect the default is the current gitea instance, if the actions isn't mirrored look in gitea.com, then github.com ( Preferable a configurable list of instances to look for actions).

Workflow and Actions files syntax compatible will bring many benefits for Gitea but I couldn't find the benefit of protocol compatible. As well we have to spend much time to research it since it's not a standard protocol and it maybe changed sometime by Github. A new protocol would make creating a public standard protocol possible for Gitea's internal or external CI/CD system.

Reply:
> > Imho all this configuration should really go into agent/bot/runner side not in gitea
> 
> But when you want to change these properties, you have to change them from runner one by one.

But what if I want to add different runners for different organizations with other configuration? If my runner/bot configuration would be added here theoretically this configuration could be added there per agent group and stored in agent database record as additional json coulmn

Reply:
> > > Imho all this configuration should really go into agent/bot/runner side not in gitea
> > 
> > 
> > But when you want to change these properties, you have to change them from runner one by one.
> 
> But what if I want to add different runners for different organizations with other configuration? If my runner/bot configuration would be added here theoretically this configuration could be added there per agent group and stored in agent database record as additional json coulmn

That could be a configuration on Gitea side per repository/per organization/per runner. And you can change it from Gitea's UI. But the global default value will be in the app.ini

Reply:
> That could be a configuration on Gitea side per repository/per organization/per runner. And you can change it from Gitea's UI. But the global default value will be in the app.ini

But why not just have this set it when adding/configuring global runner? The less it has in app.ini the easier it is to configure it, especially such specific options

Reply:
> We tried, but unfortunately, it didn't work because of the different API path, you can find additional commits in fork actions, like https://gitea.com/actions/setup-go/commit/8c779dd49b301bda581f79473e48dce4b70abe7e.

Looking at this commit, I see that instead of being hard-coded to `api.github.com`, it is hard-coded to `gitea.com/api/v1/`. So how can it work on a repo not hosted on gitea.com :thinking: ?

Reply:
> > We tried, but unfortunately, it didn't work because of the different API path, you can find additional commits in fork actions, like https://gitea.com/actions/setup-go/commit/8c779dd49b301bda581f79473e48dce4b70abe7e.
> 
> Looking at this commit, I see that instead of being hard-coded to `api.github.com`, it is hard-coded to `gitea.com/api/v1/`. So how can it work on a repo not hosted on gitea.com 🤔 ?

Yes, we should use `GITHUB_API_URL` or something that env there, a PR is welcome.